src/stores/contacts.ts

```diff
@@ -0,0 +1,25 @@
+import { defineStore } from 'pinia';
+import { useStorage } from '@vueuse/core';
+import type { Contact } from '@/types';
+import pkg from '../../package.json';
+
+export const useContactsStore = defineStore('contacts', {
+  state: () => ({
+    contacts: useStorage(`${pkg.name}.contacts`, [] as Contact[])
+  }),
+  actions: {
+    saveContact(payload: Contact) {
+      const contact = this.contacts.find(contact => contact.address === payload.address);
+      if (contact) {
+        Object.entries(payload).map(([key, value]) => {
+          contact[key] = value;
+        });
+      } else {
+        this.contacts.unshift(payload);
+      }
+    },
+    deleteContact(address: string) {
+      this.contacts = this.contacts.filter(contact => contact.address !== address);
+    }
+  }
```

Comment:
I wonder, storing them as Record<address, Contact> would make handling those actions simpler. What do you think?

Reply:
Maybe it would be easier here, but then we need somehow to sort contacts, now we unshift new contact and its easy - new contacts are at the top. And if we will add some more properties to contact it would be easier )